pkg/fanal/analyzer/language/nodejs/yarn/yarn.go

```diff
@@ -169,11 +169,54 @@ func (a yarnAnalyzer) parsePackageJsonDependencies(fsys fs.FS, path string) (map
 	}
 	defer func() { _ = f.Close() }()
 
-	pkg, err := a.packageJsonParser.Parse(f)
+	rootPkg, err := a.packageJsonParser.Parse(f)
 	if err != nil {
 		return nil, xerrors.Errorf("parse error: %w", err)
 	}
 
 	// Merge dependencies and optionalDependencies
-	return lo.Assign(pkg.Dependencies, pkg.OptionalDependencies), nil
+	dependencies := lo.Assign(rootPkg.Dependencies, rootPkg.OptionalDependencies)
+
+	if rootPkg.Workspaces != nil {
+		pkgs, err := a.traverseWorkspaces(fsys, rootPkg.Workspaces)
+		if err != nil {
+			return nil, xerrors.Errorf("traverse workspaces error: %w", err)
+		}
+		for _, pkg := range pkgs {
+			dependencies = lo.Assign(dependencies, pkg.Dependencies, pkg.OptionalDependencies)
+		}
+	}
+
+	return dependencies, nil
+}
+
+func (a yarnAnalyzer) traverseWorkspaces(fsys fs.FS, workspaces []string) ([]packagejson.Package, error) {
+	var pkgs []packagejson.Package
+
+	required := func(path string, _ fs.DirEntry) bool {
+		return filepath.Base(path) == types.NpmPkg
```

Comment:
Where is `node_modules` created? Only root?

- monorepo
    - package.json
    - node_modules <- only root?
    - c
        - package.json
        - node_modules <- ?
    - packages
        - package1
            - package.json
            - node_modules <- ?
        - package2
            - package.json
            - node_modules <- ?

Should we skip `package.json` under `node_modules` just in case?

```suggestion
		// Skip package.json files under node_modules
		if slices.Contains(strings.Split(path, "/"), "node_modules") {
			return false
		}
		return filepath.Base(path) == types.NpmPkg
```

Reply:
`node_modules` is created only at the root

Reply:
It is ok if you're sure irrelevant package.json files must not be present in workspace projects. Since we use `WalkDir`, all package.json files will be parsed under the workspace paths. That is my concern.

Reply:
We can extract the names of all the packages of the monorepository from `yarn.lock` and then check them so as not to capture an extra package.json. But as far as I know, at the moment `go-dep-parser` does not include local packages from workspaces in the list of dependencies.

Reply:
OK. I might be too concerned. If we see any issue, we can fix it.